src/proc/formats-converter.cpp

```diff
@@ -57,6 +59,24 @@ void formats_converter::drop_non_basic_formats()
     }
 }
 
+std::ostream & operator<<( std::ostream & os, const std::shared_ptr< stream_profile_interface > & profile )
```

Comment:
This is a very useful function and we keep re-implementing it in one way or another all over the place.
I suggest it be placed in a central place and not just here.
`src/core/streaming.h` is where `stream_profile_interface` is defined, I suggest the signature there and the body in a .cpp somwhere.

Reply:
p.s. we have to choose whether the syntax would be:
`... << stm_ptr;`
or
`... << *stm_ptr;`
I.e., do we pass the (SMART) POINTER to the stream interface, or a reference to it? I think we're not always going to have a smart ptr. I vote for a reference, if possible.

Reply:
A reference will be problematic since it is an abstract class. Maybe we can pass a raw pointer but I don't think that we use it.

Reply:
A general operator for `stream_profile_interface` is implemented [here](https://github.com/IntelRealSense/librealsense/blob/master/src/core/streaming.h#L83). It does not print all the details I wanted so I added specifically here. Do you think we need it also elsewhere? No other implementations/uses that I was able to find.

Reply:
> A reference will be problematic since it is an abstract class. Maybe we can pass a raw pointer but I don't think that we use it.

A ref is just syntax, it doesn't mean anything. You can take its address, compare it to null, etc., same as any pointer.

Reply:
> A general operator for `stream_profile_interface` is implemented [here](https://github.com/IntelRealSense/librealsense/blob/master/src/core/streaming.h#L83).

That's for `stream_profiles`. I vote for yours, and change that one to use it.